network/transport/v2/handlers.go

```diff
@@ -139,6 +142,31 @@ func (p *protocol) handleTransactionPayload(msg *TransactionPayload) error {
 	return p.payloadScheduler.Finished(ref)
 }
 
+func (p protocol) handleTransactionRangeQuery(peer transport.Peer, envelope *Envelope_TransactionRangeQuery) error {
+	msg := envelope.TransactionRangeQuery
+	cid := conversationID(msg.ConversationID)
+	if err := p.cMan.check(envelope); err != nil {
```

Comment:
Only the response needs to be checked. All TXs in the response must have a LC value within the requested range.

Reply:
But shouldn't TransactionRangeQuery only be sent after receiving a TransactionSet? Is that something we'd want to make explicit?

Reply:
The idea is that any tx received is a result of a request of the receiving node. How it came to that request is less relevant. The reason behind this is to limit unwanted transactions from the past (because this is less performant). If those weren't requested an evil node could push old txs to a node resulting in poor performance.

expanding conversations to be very strict would make it more complex while adding no extra security (I think)